app/scripts/lib/multichain-api/caip25permissions.test.ts

```diff
@@ -215,5 +234,397 @@ describe('endowment:caip25', () => {
+              type: Caip25CaveatType,
+              value: {
+                requiredScopes: {
+                  'eip155:1': {
+                    methods: ['eth_chainId'],
+                    notifications: [],
+                    accounts: ['eip155:1:0xdead'],
+                  },
+                },
+                optionalScopes: {
+                  'eip155:5': {
+                    methods: [],
+                    notifications: [],
+                    accounts: ['eip155:5:0xbeef'],
+                  },
+                },
+                isMultichainOrigin: true,
+              },
+            },
+          ],
+          date: 1234,
+          id: '1',
+          invoker: 'test.com',
+          parentCapability: Caip25EndowmentPermissionName,
+        });
+      } catch (err) {
+        // noop
+      }
+      expect(MockScope.assertScopesSupported).toHaveBeenCalledWith(
+        'flattenedRequiredScopes',
+        expect.objectContaining({
+          isChainIdSupported: expect.any(Function),
+        }),
+      );
+      const isChainIdSupportedBody =
+        MockScope.assertScopesSupported.mock.calls[0][1].isChainIdSupported.toString();
+      expect(isChainIdSupportedBody).toContain('findNetworkClientIdByChainId');
+    });
+
+    it('asserts the validated and flattened optional scopes are supported', () => {
+      MockScope.validateAndFlattenScopes.mockReturnValue({
+        flattenedRequiredScopes: 'flattenedRequiredScopes',
+        flattenedOptionalScopes: 'flattenedOptionalScopes',
+      });
+      try {
+        validator({
+          caveats: [
+            {
+              type: Caip25CaveatType,
+              value: {
+                requiredScopes: {
+                  'eip155:1': {
+                    methods: ['eth_chainId'],
+                    notifications: [],
+                    accounts: ['eip155:1:0xdead'],
+                  },
+                },
+                optionalScopes: {
+                  'eip155:5': {
+                    methods: [],
+                    notifications: [],
+                    accounts: ['eip155:5:0xbeef'],
+                  },
+                },
+                isMultichainOrigin: true,
+              },
+            },
+          ],
+          date: 1234,
+          id: '1',
+          invoker: 'test.com',
+          parentCapability: Caip25EndowmentPermissionName,
+        });
+      } catch (err) {
+        // noop
+      }
+      expect(MockScope.assertScopesSupported).toHaveBeenCalledWith(
+        'flattenedOptionalScopes',
+        expect.objectContaining({
+          isChainIdSupported: expect.any(Function),
+        }),
+      );
+      const isChainIdSupportedBody =
+        MockScope.assertScopesSupported.mock.calls[1][1].isChainIdSupported.toString();
+      expect(isChainIdSupportedBody).toContain('findNetworkClientIdByChainId');
+    });
+
+    it('throws if the input requiredScopes ScopesObject is not already validated and flattened', () => {
```

Comment:
Should there be requiredScopes in the mocked value returned by the `validateAndFlattenScopes` mock for this test?

Reply:
see comment below https://github.com/MetaMask/metamask-extension/pull/26698#discussion_r1735351718